test/new-e2e/pkg/environments/aws/ecs/ecs.go

```diff
@@ -0,0 +1,290 @@
+	if env.AwsEnvironment != nil {
+		awsEnv = *env.AwsEnvironment
+	} else {
+		awsEnv, err = aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+	}
+	// Create cluster
+	ecsCluster, err := ecs.CreateEcsCluster(awsEnv, params.name)
+	if err != nil {
+		return err
+	}
+
+	// Export cluster’s properties
+	ctx.Export("ecs-cluster-name", ecsCluster.Name)
+	ctx.Export("ecs-cluster-arn", ecsCluster.Arn)
+	env.ClusterName = ecsCluster.Name
+	env.ClusterArn = ecsCluster.Arn
+
+	// Handle capacity providers
+	capacityProviders := pulumi.StringArray{}
+	if params.ecsFargate {
+		capacityProviders = append(capacityProviders, pulumi.String("FARGATE"))
+	}
+
+	linuxNodeGroupPresent := false
+	if params.ecsLinuxECSOptimizedNodeGroup {
+		cpName, err := ecs.NewECSOptimizedNodeGroup(awsEnv, ecsCluster.Name, false)
+		if err != nil {
+			return err
+		}
+
+		capacityProviders = append(capacityProviders, cpName)
+		linuxNodeGroupPresent = true
+	}
+
+	if params.ecsLinuxECSOptimizedARMNodeGroup {
+		cpName, err := ecs.NewECSOptimizedNodeGroup(awsEnv, ecsCluster.Name, true)
+		if err != nil {
+			return err
+		}
+
+		capacityProviders = append(capacityProviders, cpName)
+		linuxNodeGroupPresent = true
+	}
+
+	if params.ecsLinuxBottlerocketNodeGroup {
+		cpName, err := ecs.NewBottlerocketNodeGroup(awsEnv, ecsCluster.Name)
+		if err != nil {
+			return err
+		}
+
+		capacityProviders = append(capacityProviders, cpName)
+		linuxNodeGroupPresent = true
+	}
+
+	if params.ecsWindowsNodeGroup {
+		cpName, err := ecs.NewWindowsNodeGroup(awsEnv, ecsCluster.Name)
+		if err != nil {
+			return err
+		}
+
+		capacityProviders = append(capacityProviders, cpName)
+	}
+
+	// Associate capacity providers
+	_, err = ecs.NewClusterCapacityProvider(awsEnv, ctx.Stack(), ecsCluster.Name, capacityProviders)
+	if err != nil {
+		return err
+	}
+
+	var apiKeyParam *ssm.Parameter
+	var fakeIntake *fakeintakeComp.Fakeintake
+	// Create task and service
+	if params.agentOptions != nil {
+		if params.fakeintakeOptions != nil {
+			fakeIntakeOptions := []fakeintake.Option{}
+			fakeIntakeOptions = append(fakeIntakeOptions, params.fakeintakeOptions...)
+			if awsEnv.InfraShouldDeployFakeintakeWithLB() {
+				fakeIntakeOptions = append(fakeIntakeOptions, fakeintake.WithLoadBalancer())
+			}
+
+			if fakeIntake, err = fakeintake.NewECSFargateInstance(awsEnv, "ecs", fakeIntakeOptions...); err != nil {
+				return err
+			}
+			if err := fakeIntake.Export(awsEnv.Ctx, &env.FakeIntake.FakeintakeOutput); err != nil {
+				return err
+			}
+		}
+		apiKeyParam, err = ssm.NewParameter(ctx, awsEnv.Namer.ResourceName("agent-apikey"), &ssm.ParameterArgs{
+			Name:      awsEnv.CommonNamer.DisplayName(1011, pulumi.String("agent-apikey")),
+			Type:      ssm.ParameterTypeSecureString,
+			Overwrite: pulumi.Bool(true),
+			Value:     awsEnv.AgentAPIKey(),
+		}, awsEnv.WithProviders(config.ProviderAWS))
+		if err != nil {
+			return err
+		}
+
+		// Deploy EC2 Agent
+		if linuxNodeGroupPresent {
```

Comment:
What happens if this is `false`? Do we not have an agent running for non-Linux?

Reply:
for the moment the agent is not deployed on ecsWindowsNodeGroup but this where the else code will go